packages/app-store/stripepayment/api/paymentCallback.ts

```diff
@@ -43,10 +45,10 @@ async function getHandler(req: NextApiRequest, res: NextApiResponse) {
     });
   }
 
+  if (!user) throw new HttpError({ statusCode: 404, message: "User not found" });
+
   if (checkoutSession.payment_status === "paid" && stripeCustomer.metadata.username) {
     try {
-      if (!user) return { message: "User not found" };
-
```

Comment:
did this make sense ? why did we check for user only if payment_status was paid and stripeCustomer.metadata.username was set ?

Reply:
does it make sense to redirect a user that is not in the database ?

Reply:
This could have just been to silence typescript. When commenting it out we get typescript errors. The chances of no user being found are very very slim. Before sending the user to the checkout page we get the userId from Stripe. We then get the query again by the Stripe customer. I highly doubt that that information would change during the time the user is checking out.

Reply:
Looks good to me.